### PR TITLE
[DOCS] Improve PHP doc comments

### PR DESCRIPTION
**Pull request by linawolf**
* AbstractTcaRector::removeArrayItemFromArrayByKey

### CODE REVIEW

File: src/Rector/AbstractTcaRector.php [hunk 1]
```diff
@@ -337,6 +337,26 @@ protected function hasKeyValuePair(Array_ $configValueArray, string $configKey,
...
+     *
+     * ```
+     * $appearanceArray = $this->extractSubArrayByKey($configArray, 'appearance');
+     *      if (! $appearanceArray instanceof Array_) {
+     *      return;
+     * }
+     * $this->removeArrayItemFromArrayByKey($appearanceArray, 'showRemovedLocalizationRecords');
+     * ```
+     *
+     * Attention: Strict comparizion is used for the key. key with int 42 will
```
Comment (sabbelasichon) [inline src/Rector/AbstractTcaRector.php:357]:
```suggestion
     * Attention: Strict comparison is used for the key. key with int 42 will
```

File: src/Rector/AbstractTcaRector.php [hunk 2]
```diff
@@ -337,6 +337,26 @@ protected function hasKeyValuePair(Array_ $configValueArray, string $configKey,
...
+     * ```
+     * $this->removeArrayItemFromArrayByKey($configArray, 'myKeyToBeRemoved');
+     * ```
+     *
+     * If the key to be removed is in a sub array of the current one
+     * use `extractSubArrayByKey` to extract the sub array first:
+     *
+     * ```
+     * $appearanceArray = $this->extractSubArrayByKey($configArray, 'appearance');
+     *      if (! $appearanceArray instanceof Array_) {
```
Comment (simonschaufi) [inline src/Rector/AbstractTcaRector.php:351]: indentation is wrong

File: src/Rector/AbstractTcaRector.php [hunk 3]
```diff
@@ -337,6 +337,26 @@ protected function hasKeyValuePair(Array_ $configValueArray, string $configKey,
...
+     *      return;
```
Comment (linawolf) [inline src/Rector/AbstractTcaRector.php:352]:
```suggestion
     * $appearanceArray = $this->extractSubArrayByKey($configArray, 'appearance');
     *      if (! $appearanceArray instanceof Array_) {
     *          return;
```

File: src/Rector/AbstractTcaRector.php (hunk 2 above)
Comment (simonschaufi) [reply src/Rector/AbstractTcaRector.php:351]: yes but the if needs to start at the beginning of the line.

Comment (simonschaufi) [inline src/Rector/AbstractTcaRector.php:351]:
```suggestion
     * if (! $appearanceArray instanceof Array_) {
```

File: src/Rector/AbstractTcaRector.php [hunk 4]
```diff
@@ -337,6 +337,26 @@ protected function hasKeyValuePair(Array_ $configValueArray, string $configKey,
...
+     * $this->removeArrayItemFromArrayByKey($configArray, 'myKeyToBeRemoved');
+     * ```
+     *
+     * If the key to be removed is in a sub array of the current one
+     * use `extractSubArrayByKey` to extract the sub array first:
+     *
+     * ```
+     * $appearanceArray = $this->extractSubArrayByKey($configArray, 'appearance');
+     * if (! $appearanceArray instanceof Array_) {
+     *          return;
```
Comment (simonschaufi) [inline src/Rector/AbstractTcaRector.php:352]:
```suggestion
     *     return;
```

**Comment by linawolf** [comment]
Sorry about that, not feeling well and forgot to follow up. Thanks for taking care

**Pull request merged**